opensearch-operator/pkg/builders/cluster.go

```diff
@@ -852,3 +859,56 @@ func DataNodesCount(ctx context.Context, k8sClient client.Client, cr *opsterv1.O
 	}
 	return count
 }
+
+func NewServiceMonitor(cr *opsterv1.OpenSearchCluster) *prometheus.ServiceMonitor {
+
+	labels := map[string]string{
+		ClusterLabel: cr.Name,
+	}
+	selector := metav1.LabelSelector{
+		MatchLabels: labels,
+	}
+
+	namespaceSelector := prometheus.NamespaceSelector{
+		Any:        false,
+		MatchNames: []string{cr.Namespace},
+	}
+	return &prometheus.ServiceMonitor{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      cr.ClusterName + "-monitor",
+			Namespace: cr.Namespace,
+			Labels:    labels,
+		},
+		Spec: prometheus.ServiceMonitorSpec{
+			JobLabel: cr.ClusterName + "-monitor",
+			TargetLabels: []string{
+				cr.ClusterName,
+			},
+			PodTargetLabels: []string{
+				cr.ClusterName,
+			},
+			Endpoints: []prometheus.Endpoint{
+				{Port: "9200",
+					TargetPort:      nil,
+					Path:            "/_prometheus/metrics",
+					Interval:        "10s",
```

Comment:
I think `10s` is too often. From my perspective `30s` is a good default. Maybe make this configurable for the user?

Reply:
Will make it configurable